eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -278,4 +346,116 @@ where
         res.truncate_to_canonical_form();
         res
     }
+
+    pub fn plaintext_add(&self, m1: Message<C>, m2: Message<C>) -> Message<C> {
+        let mut res = m1.m + m2.m;
+        for coeff in res.coeffs_mut() {
+            let mut coeff_res = C::coeff_as_u128(*coeff);
+            coeff_res %= C::t_as_u128();
+            *coeff = coeff_res.into();
+        }
+        res.truncate_to_canonical_form();
+        Message { m: res }
+    }
+
+    pub fn plaintext_mul(&self, m1: Message<C>, m2: Message<C>) -> Message<C> {
+        let mut res = m1.m * m2.m;
+        for coeff in res.coeffs_mut() {
+            let mut coeff_res: i128 = C::coeff_as_u128(*coeff) as i128;
+            // center lift mod q
+            if coeff_res > (C::modulus_minus_one_div_two_as_u128() as i128) {
+                coeff_res -= C::modulus_as_u128() as i128;
+            }
+            coeff_res %= C::t_as_u128() as i128;
+            if coeff_res < 0 {
+                coeff_res += C::t_as_u128() as i128;
+            }
```

Comment:
This is a manual implementation of `rem_euclid()`:
https://doc.rust-lang.org/std/primitive.i128.html#method.rem_euclid

```suggestion
            coeff_res = coeff_res.rem_euclid(C::t_as_u128() as i128);
```

Reply:
done